azurerm/resource_arm_storage_account_blob_settings.go

```diff
@@ -0,0 +1,177 @@
+
+	resourceGroupName := d.Get("resource_group_name").(string)
+	storageAccountName := d.Get("storage_account_name").(string)
+	enableSoftDelete := d.Get("enable_soft_delete").(bool)
+	softDeleteRetentionDays := int32(d.Get("soft_delete_retention_days").(int))
+
+	if requireResourcesToBeImported && d.IsNewResource() {
+		existing, err := client.GetServiceProperties(ctx, resourceGroupName, storageAccountName)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("error checking for presence of existing Storage Account Blob Settings (Storage Account %q / Resource Group %q): %s",
+					storageAccountName, resourceGroupName, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError(storageAccountBlobSettingsResourceName, *existing.ID)
+		}
+	}
+
+	properties := storage.BlobServicePropertiesProperties{
+		DeleteRetentionPolicy: &storage.DeleteRetentionPolicy{
+			Enabled: &enableSoftDelete,
+			Days:    &softDeleteRetentionDays,
+		},
+	}
+
+	propertiesWrapper := storage.BlobServiceProperties{
+		BlobServicePropertiesProperties: &properties,
+	}
+
+	_, err := client.SetServiceProperties(ctx, resourceGroupName, storageAccountName, propertiesWrapper)
+
+	if err != nil {
+		return fmt.Errorf("error while updataing Storage Account Blob Settings (Storage Account %q / Resource Group %q): %s",
+			storageAccountName, resourceGroupName, err)
+	}
+
+	read, err := client.GetServiceProperties(ctx, resourceGroupName, storageAccountName)
+
+	if err != nil || read.ID == nil {
+		return fmt.Errorf("can't read ID of Storage Account Blob Settings (Storage Account %q / Resource Group %q): %s",
+			storageAccountName, resourceGroupName, err)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmStorageAccountBlobSettingsRead(d, meta)
+}
+
+func resourceArmStorageAccountBlobSettingsRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).storageBlobServicesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] Reading Azure ARM Storage Account Blob Settings.")
+
+	id, err := parseAzureResourceID(d.Id())
+
+	if err != nil {
+		return err
+	}
+
+	resourceGroupName := id.ResourceGroup
+	storageAccountName := id.Path["storageAccounts"]
+
+	resp, err := client.GetServiceProperties(ctx, resourceGroupName, storageAccountName)
+
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("error making Read request on Azure Storage Account %q: %+v", storageAccountName, err)
+	}
+
+	d.Set("resource_group_name", resourceGroupName)
+	d.Set("storage_account_name", storageAccountName)
+
+	if props := resp.BlobServicePropertiesProperties; props != nil {
+		if policy := props.DeleteRetentionPolicy; policy != nil {
+			if policy.Enabled != nil {
+				d.Set("enable_soft_delete", policy.Enabled)
+			}
+			if policy.Days != nil {
+				d.Set("soft_delete_retention_days", policy.Days)
+			}
```

Comment:
(as above) `d.Set` will handle this being a nilable pointer (for simple types) as such we can remove the if statement and make this:

```suggestion
			d.Set("soft_delete_retention_days", policy.Days)
```

Reply:
done